ipc/src/server.rs

```diff
@@ -104,11 +105,11 @@ impl<M: Metadata, S: Middleware<M>> ServerBuilder<M, S> {
 		self
 	}
 
-	/// Run server (in a separate thread)
-	pub fn start(self, path: &str) -> std::io::Result<Server> {
+	/// Creates a new server from the given endpoint.
+	pub fn from_endpoint(self, endpoint: Endpoint) -> std::io::Result<Server> {
```

Comment:
I don't like the fact that we require some external type here.

I think we should:
1. Leave `start` as is
2. Expose `pipe_security` (or sth similar) method on the `ServerBuilder`.
3. Keep `parity-tokio-ipc` as implementation detail.

Reply:
That would involve keeping a `*mut SECURITY_ATTRIBUTES` field in the server builder. The whole situation around Windows ACL APIs for now is a bit messy because there isn't a nice Rust wrapper around it. `winapi` is the only thing that I'm aware of for now. If you believe that the pointer should still be stored on the server builder, I can go ahead and implement it that way!

Reply:
`parity-tokio-ipc` is supposed to be an abstraction for Unix and Windows sockets, in JSONRPC-IPC server we don't really care about the platform or even about particular permissions. As Niko suggested in https://github.com/NikVolf/parity-tokio-ipc/pull/6#discussion_r212315188 we should have a safe wrapper around the API anyway.

In server builder we could even have a method: `allow_unprivileged_access()` that doesn't take any parameters but just sets the required flags.

Reply:
Well, I've updated both `parity-tokio-ipc` and this PR. I think this one deserves to be renamed as it has nothing to do with exposing a way to construct a server from an endpoint.

My original intention was to allow to separate the server code further away from the internals of the `parity-tokio-ipc::Endpoint`, as to allow to keep the platform specific stuff inside the Endpoint. Currently, the server implementation has to be _platform aware_.